### PR TITLE
Move default data folder for *nix platforms to $XDG_DATA_HOME/jellyfin

### PR DESCRIPTION
**Pull request by Bond-009**
Remove dependency on System.Configuration.ConfigurationManager

### CODE REVIEW

**Comment by Bond-009** [comment]
`$XDG_CONFIG_HOME defines the base directory relative to which user specific configuration files should be stored`
It's isn't a folder for configuration files

**Comment by nvllsvm** [comment]
What is stored here?

**Comment by Bond-009** [comment]
`cache config data ffmpeg localization logs root`

**Comment by nvllsvm** [comment]
It should respect `XDG_DATA_HOME` then. 

I'm being particularly picky about this since it's the type of thing we need to get right once. Any changes could require user interaction.

**Comment by joshuaboniface** [comment]
I completely agree there - can you fix that up @Bond-009 ?

**Comment by joshuaboniface** [comment]
@JustAMan yes we override it in the command in the Debian package using `-programdata`.

File: MediaBrowser.Server.Mono/Program.cs [hunk 1]
```diff
@@ -87,12 +88,26 @@ private static ServerApplicationPaths CreateApplicationPaths(string applicationP
...
+                    // $XDG_DATA_HOME defines the base directory relative to which user specific data files should be stored.
+                    programDataPath = Environment.GetEnvironmentVariable("XDG_DATA_HOME");
+                    // If $XDG_DATA_HOME is either not set or empty, $HOME/.local/share should be used.
+                    if (string.IsNullOrEmpty(programDataPath)){
+                        programDataPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), ".local", "share");
+                    }
+                }
             }
 
+            programDataPath = Path.Combine(programDataPath, "jellyfin");
```
Comment (joshuaboniface) [inline MediaBrowser.Server.Mono/Program.cs]: Can we add this as a conditional if the programDataPath doesn't already contain the string `jellyfin`? This results in duplicate directory structures in the Debian build:

```
$ ps aux | grep jellyfin
jellyfin  3676  0.8  4.9 3488040 100916 ?      SLsl 16:36   0:06 /usr/lib/jellyfin/bin/jellyfin -programdata /var/lib/jellyfin -restartpath /usr/lib/jellyfin/restart.sh
```
```
$ ls /var/lib/jellyfin/jellyfin/
cache  config  data  ffmpeg  localization  logs  root
```

Comment (joshuaboniface) [reply MediaBrowser.Server.Mono/Program.cs]: >basically, if the user wants to be explicit we should let them and not create a second level subdir

Fixed in #159.

**Pull request merged**